src/hotspot/share/gc/shared/c2/barrierSetC2.hpp

```diff
@@ -92,9 +92,9 @@ class C2AccessValuePtr: public C2AccessValue {
 
 public:
   C2AccessValuePtr(Node* node, const TypePtr* type) :
-    C2AccessValue(node, reinterpret_cast<const Type*>(type)) {}
+    C2AccessValue(node, type) {}
 
-  const TypePtr* type() const { return reinterpret_cast<const TypePtr*>(_type); }
+  const TypePtr* type() const { return static_cast<const TypePtr*>(_type); }
```

Comment:
I think this should be:

```suggestion
  const TypePtr* type() const { return _type->is_ptr(); }
```

Which has the advantage that we assert that the type is as expected.

Reply:
And yes the name of this method is suboptimal :)